packages/stencil/src/components/datacom-pagination/datcom-pagination.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  Component,
+  Host,
+  h,
+  Prop,
+  Event,
+  EventEmitter,
+  Listen,
+} from '@stencil/core';
+import { getSvg } from '../../common/images';
+
+@Component({
+  tag: 'datacom-pagination',
+  styleUrl: 'datacom-pagination.css',
+  scoped: true,
+})
+export class DatacomPagination {
+  @Prop({ mutable: true }) itemsPerPage?: number;
+  @Prop({ mutable: true }) totalItems?: number;
+  @Prop({ mutable: true }) currentPage?: number;
+  @Event({
+    composed: true,
+  })
+  private pageChanged: EventEmitter<number>;
+  // private inputElement: HTMLInputElement;
+
+  /**
+   * Handle boundary cases such as only allowing to navigate between the range.
+   */
+  private goToPage(pageNumber: number) {
+    if (pageNumber < 1 || pageNumber > this.totalPages) {
+      return;
+    }
+    this.currentPage = pageNumber;
+    this.pageChanged.emit(this.currentPage);
+  }
+
+  /**
+   * calculate the total number of pages based on the total number of items and the items per page.
+   */
+  get totalPages(): number {
+    return Math.ceil(this.totalItems / this.itemsPerPage);
+  }
+
+  onClickFirstPageHandler = () => {
+    this.goToPage(1);
+    this.pageChanged.emit(this.currentPage);
+  };
+
+  onClickPerviousPageHandler = () => {
+    this.goToPage(this.currentPage - 1);
+    this.pageChanged.emit(this.currentPage);
+  };
+
+  onClickNextPageHandler = () => {
+    this.goToPage(+this.currentPage + 1);
+    this.pageChanged.emit(this.currentPage);
+  };
+
+  onClickLastPageHandler = () => {
+    this.currentPage = this.totalPages;
+    this.pageChanged.emit(this.currentPage);
+  };
+
+  /**
+   * Update the value when enter the page number in the input field.
+   */
+  onPageChangeHandler = (event: Event): void => {
```

Comment:
Please change the parameter typing from `Event` to `KeyboardEvent` | `MouseEvent`